Airwallex/Card/Internal/AWXCardViewController.swift

```diff
@@ -0,0 +1,669 @@
+        dismiss(animated: true) {
+            let delegate = AWXUIContext.shared().delegate
+            delegate?.paymentViewController(self, didCompleteWith: .cancel, error: nil)
+            self.logMessage(
+                "Delegate: \(delegate?.description ?? ""), paymentViewController:didCompleteWithStatus:error: \(self.presentationController?.description ?? "")"
+            )
+        }
+    }
+}
+
+extension AWXCardViewController: AWXFloatingLabelTextFieldDelegate {
+    public func floatingLabelTextField(
+        _: AWXFloatingLabelTextField,
+        textFieldShouldBeginEditing textField: UITextField
+    ) -> Bool {
+        switch textField {
+        case cardNoField.textField:
+            Risk.log(event: "input_card_number", screen: "page_create_card")
+        case cvcField.textField:
+            Risk.log(event: "input_card_cvc", screen: "page_create_card")
+        case expiresField.textField:
+            Risk.log(event: "input_card_expiry", screen: "page_create_card")
+        case nameField.textField:
+            Risk.log(event: "input_card_holder_name", screen: "page_create_card")
+        default:
+            break
+        }
+        return true
+    }
+}
+
+extension AWXCardViewController: AWXCountryListViewControllerDelegate {
+    public func countryListViewController(
+        _ controller: AWXCountryListViewController, didSelect country: AWXCountry
+    ) {
+        controller.dismiss(animated: true)
+        viewModel?.selectedCountry = country
+        countryView.setText(country.countryName, animated: false)
+    }
+}
+
+extension AWXCardViewController {
+    func registerKeyboard() {
+        NotificationCenter.default.addObserver(
+            self, selector: #selector(keyboardWillChangeFrame(_:)),
+            name: UIResponder.keyboardWillChangeFrameNotification, object: nil
+        )
+        NotificationCenter.default.addObserver(
+            self, selector: #selector(keyboardWillBeHidden(_:)),
+            name: UIResponder.keyboardWillHideNotification, object: nil
+        )
+    }
+
+    func unregisterKeyboard() {
+        NotificationCenter.default.removeObserver(
+            self, name: UIResponder.keyboardWillChangeFrameNotification, object: nil
+        )
+        NotificationCenter.default.removeObserver(
+            self, name: UIResponder.keyboardWillHideNotification, object: nil
+        )
+    }
+
+    func keyboardWillChangeFrame(_ notification: Notification) {
+        if let rect = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect {
+            let keyboardHeight = CGRectGetHeight(rect)
+            scrollView.contentInset.bottom = keyboardHeight
+            scrollView.verticalScrollIndicatorInsets.bottom = keyboardHeight
+        }
+    }
+
+    func keyboardWillBeHidden(_: Notification) {
+        scrollView.contentInset = .zero
+        scrollView.verticalScrollIndicatorInsets = .zero
+    }
+}
+
+extension AWXCardViewController: AWXProviderDelegate {
```

Comment:
I feel this AWXCardViewController should conform to `AWXCardViewModelDelegate` which only contains necessary presentation methods (e.g. `shouldPresentVC`, `shouldInsertVC`, `startLoading`, `stopLoading`), and provider should be moved to viewModel instead.